solr/core/src/java/org/apache/solr/cloud/overseer/ReplicaMutator.java

```diff
@@ -438,9 +438,6 @@ private ZkWriteCommand updateState(
 
     DocCollection newCollection = CollectionMutator.updateSlice(collectionName, collection, slice);
     log.debug("Collection is now: {}", newCollection);
-    if (collection.isPerReplicaState() && oldReplica != null) {
```

Comment:
Node publishes the state of shard-active and then it updates the prs state of replica. We need to fetch the prs state in overseer node. By removing code, overseer node fetches the prs state in `writePendingState` method. That fixes this issue.

(But definitely there is scope to refactor overseer code )